added a copy button which copies all the errors and traceback to clip…

Mr-Rahul-Paul opened the pull request:
Fixes #25599

This PR fixes #25599 by adding a "Copy" button to the Unit Test dialog, allowing users to copy the details of failing tests directly to the clipboard.

### Before
No easy way to extract the list of Errors from the UI.
<img width="1390" height="1220" alt="image" src="https://github.com/user-attachments/assets/83b1f543-55f4-4049-8b69-1f1e6fc0c5b1" />

### After
A new "**Copy**" button is added **below** the Close button. On clicking,  it copies errors and traceback(s) to the system clipboard.
<img width="463" height="457" alt="image" src="https://github.com/user-attachments/assets/e5873673-83f5-480f-b70b-897b3f24bbdd" />


### How I reproduced the Issue
1. Open the demo file AssemblyExample.FCStd.
2. Enable the Test Framework workbench from Preferences.
3. Run Self-test → TestApp.All.
4. Click Start and wait for failures to appear in the list.

## Changes done

-  Added "Copy" button in `UnitTest.ui` file with name "copyButton"
- Declared `onCopyButtonClicked()` function in `UnitTestImp.h`
- clicking 'copybutton' calls 'onCopyButtonClicked()' which copies the error(s) to clipboard

## Remarks
- the logic used in `onCopyButtonClicked()` is that function iterates through the list of failures (`treeViewFailure`), extracting both the error names and the tracebacks, appending them to a string variable which is then copied to the clipboard.
- I have tried my best to ensure all new function names (like `onCopyButtonClicked`) and variable naming conventions follow the existing patterns to maintain consistency

xtemp09 commented:
```xml
<string>&amp;Copy </string>
```



I think that the ampersand before _C_ must be removed, since it is present in the _Close_ button's text.




Why does this string end with space?

xtemp09 commented:
> > I think that the ampersand before C must be removed, since it is present in the Close button's text.
> 
> I tried to match the pattern of the code in the file `UnitTesImp.cpp` (so as not to make a mess and keep the code structure consistent)


The ampersand must be removed.




----



> I am also a freelance 3d artist. So I based it on Blender's behaviour. In Blender, there is often a box that pops up in the bottom when some certain tasks are performed, so I thought adding a pop-up to notify would be a good idea. (this can be changed if the maintainers don't like it.)


I think that a pop-up notification is the best option along with a message in the status bar. =) But, the _About_ dialog already has a button to copy to clipboard. I think it would be beneficial to mimic its behavior.



I think also that placing the _Copy_ button next to the _Close_ button is not a good idea. In my opinion, they should be separated with a separator or placed in different locations.

Mr-Rahul-Paul commented:
- Removed ampersand from 'Close'
- removed space from `copy` button 
- after copying the Errors , the message shows up in the status bar (below the error tree) instead of a popup
- removed an unwanted header

xtemp09 commented:
It appears the tab order needs to be adjusted.

chennes commented:
Thanks for the PR, @Mr-Rahul-Paul (and thanks @xtemp09 for the quick review). @FreeCAD/design-working-group any opinion on this button layout?

Mr-Rahul-Paul commented:
@chennes any feed-back from you is appreciated

chennes commented on src/Mod/Test/Gui/UnitTest.ui:
```diff
@@ -303,6 +316,7 @@
   <tabstop>helpButton</tabstop>
   <tabstop>aboutButton</tabstop>
   <tabstop>closeButton</tabstop>
+  <tabstop>copyButton</tabstop>
```
This should now be above the help button.

chennes reviewed (approved):
I've made a couple of changes
